### PR TITLE
Require VS 2019 for linking the native image

### PR DESCRIPTION
**Pull request by jkotas**
The runtime libraries are built using VS 2019

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/findvcvarsall.bat [hunk 1]
```diff
@@ -15,6 +15,7 @@ IF /I "%~1"=="arm64" SET toolsSuffix=ARM64
 FOR /F "tokens=*" %%i IN (
     '"%vswherePath%" -latest -prerelease -products * ^
     -requires Microsoft.VisualStudio.Component.VC.Tools.%toolsSuffix% ^
+    -version [16,16] ^
```
Comment (AntonLapounov) [inline src/coreclr/nativeaot/BuildIntegration/findvcvarsall.bat]:
```suggestion
    -version [16,17) ^
```
Or `[16,)`. The current version is 16.9.*.

**Review by AntonLapounov — APPROVED** [review]

File: src/coreclr/nativeaot/BuildIntegration/findvcvarsall.bat (hunk 1 above)
Comment (jkotas) [reply src/coreclr/nativeaot/BuildIntegration/findvcvarsall.bat]: Let's see whether this is going to work with cmd. I am seeing cmd behaving differently in CI vs. locally on my machine.

File: src/coreclr/nativeaot/BuildIntegration/findvcvarsall.bat [hunk 2]
```diff
@@ -15,6 +15,7 @@ IF /I "%~1"=="arm64" SET toolsSuffix=ARM64
 FOR /F "tokens=*" %%i IN (
     '"%vswherePath%" -latest -prerelease -products * ^
     -requires Microsoft.VisualStudio.Component.VC.Tools.%toolsSuffix% ^
+    -version [16,17) ^
```
Comment (AntonLapounov) [inline src/coreclr/nativeaot/BuildIntegration/findvcvarsall.bat]:
```suggestion
    -version [16^,17^) ^
```

File: src/coreclr/nativeaot/BuildIntegration/findvcvarsall.bat (hunk 1 above)
Comment (AntonLapounov) [reply src/coreclr/nativeaot/BuildIntegration/findvcvarsall.bat]: I see why.  We need to escape both `,` and `)`.

**Pull request merged**